### PR TITLE
Bump CAPI to v1.7.0

### PR DESCRIPTION
**Pull request by damdo**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

CAPI v1.7.0 is out now, so this PRs bumps the CAPI version used by CAPG to it.

Bumps CAPI to v1.7.x based on https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/migrations/v1.6-to-v1.7.md

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [x] adds or updates e2e tests

**Release note**:
```release-note
Bump CAPI to v1.7.0
```

### CODE REVIEW

**Comment by damdo** [comment]
/hold
It is a WIP for now

**Comment by damdo** [comment]
@cpanato do you know what could be causing this failure? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-gcp/1193/pull-cluster-api-provider-gcp-e2e-test/1778061302942404608

Which seems to be hitting: https://github.com/kubernetes-sigs/cluster-api/blob/4bfc838762ff79153f29a7059865867f20f19e00/cmd/clusterctl/client/repository/repository_versions.go#L127-L134

Is it because clusterctl init in this case doesn't correctly load pre-releases (e.g. beta, release.candidates etc.) ?

Thanks!

File: Makefile [hunk 1]
```diff
@@ -59,23 +59,23 @@ CONVERSION_VERIFIER:= $(TOOLS_BIN_DIR)/conversion-verifier
...
+CONVERSION_GEN_VER := v0.29.3
 CONVERSION_GEN_BIN := conversion-gen
 CONVERSION_GEN := $(TOOLS_BIN_DIR)/$(CONVERSION_GEN_BIN)-$(CONVERSION_GEN_VER)
 
 ENVSUBST_VER := v1.4.2
 ENVSUBST_BIN := envsubst
 ENVSUBST := $(TOOLS_BIN_DIR)/$(ENVSUBST_BIN)
 
-GOLANGCI_LINT_VER := v1.55.2
+GOLANGCI_LINT_VER := v1.57.1
```
Comment (cpanato) [inline Makefile:74]:
```suggestion
GOLANGCI_LINT_VER := v1.57.2
```

**Review by cpanato — CHANGES_REQUESTED** [review]
update the metadata.yaml in root as well

**Comment by cpanato** [comment]
> @cpanato do you know what could be causing this failure? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-gcp/1193/pull-cluster-api-provider-gcp-e2e-test/1778061302942404608
> 
> Which seems to be hitting: https://github.com/kubernetes-sigs/cluster-api/blob/4bfc838762ff79153f29a7059865867f20f19e00/cmd/clusterctl/client/repository/repository_versions.go#L127-L134
> 
> Is it because clusterctl init in this case doesn't correctly load pre-releases (e.g. beta, release.candidates etc.) ?
> 
> Thanks!

update the metadata.yaml in the root as well

File: metadata.yaml [hunk 2]
```diff
@@ -32,3 +32,6 @@ releaseSeries:
   - major: 1
     minor: 6
     contract: v1beta1
+  - major: 1
+    minor: 7
+    contract: v1beta1
```
Comment (damdo) [inline metadata.yaml:37]: @cpanato It's already update here

Comment (cpanato) [reply metadata.yaml:37]: ah got it

you need to update the line 57 in test/e2e/config/gcp-ci.yaml

```
 - name: v1.6.99 # next; use manifest from source files
```

should be

```
 - name: v1.7.99 # next; use manifest from source files
```

Comment (damdo) [reply metadata.yaml:37]: Looks like that fixed it 🙌 
Thank you 😊

**Review by neolit123 — COMMENTED** [review]
/approve
/remove-kind support
/kind feature

File: Tiltfile [hunk 3]
```diff
@@ -18,9 +18,9 @@ settings = {
     "deploy_cert_manager": True,
     "preload_images_for_kind": True,
     "kind_cluster_name": "capg",
-    "capi_version": "v1.6.2",
+    "capi_version": "v1.7.0",
     "cert_manager_version": "v1.11.0",
```
Comment (richardcase) [inline Tiltfile:22]: We could probably update the cert-manager to be in-line with the version automatically installed via clusterctl. But that can be a different PR.

**Comment by damdo** [comment]
Thanks Richard
/unhold

File: Tiltfile (hunk 3 above)
Comment (damdo) [reply Tiltfile:22]: Will open a separate one for this.

**Comment by cpanato** [comment]
@damdo @richardcase did we ran conformance tests for this pr?

**Comment by damdo** [comment]
@cpanato yes we did: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-gcp/1193/pull-cluster-api-provider-gcp-conformance/1780481030948392960

File: Tiltfile (hunk 3 above)
Comment (damdo) [reply Tiltfile:22]: Done here: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1210
